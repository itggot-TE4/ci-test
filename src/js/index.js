const Hello = () => {
    return(1);
}

module.exports = Hello