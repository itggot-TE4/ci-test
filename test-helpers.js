export class TestHelper {

    constructor(container) {
        this.container = container;
    }

    cardIsPresent(card) {
        return Array.from(this.container.querySelectorAll('header')).some((node) => node.textContent === card)
    };
}