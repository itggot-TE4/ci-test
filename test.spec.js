import { fireEvent, getByText, getByDisplayValue } from '@testing-library/dom'
import '@testing-library/jest-dom/extend-expect'
import { JSDOM, VirtualConsole, CookieJar } from 'jsdom'
import { TestHelper } from './test-helpers.js'
import fs from 'fs'
import path from 'path'

//const html = fs.readFileSync(path.resolve(__dirname, './index.html'), 'utf8');

let h
let dom
let container

xdescribe('index.html', () => {
  beforeEach((done) => {
    // Constructing a new JSDOM with this option is the key
    // to getting the code in the script tag to execute.
    // This is indeed dangerous and should only be done with trusted content.
    // https://github.com/jsdom/jsdom#executing-scripts
    let url = 'http://serving:8000';

    let options = {
        resources: 'usable',
        runScripts: 'dangerously'
    }


    JSDOM.fromURL(url, options).then((dom) => {
        setTimeout(() => {
            container = dom.window.document.body
            h = new TestHelper(container)
            done();
        }, 500);
    })
  })

  it('has a main', () => {
    expect(container.querySelector('main')).not.toBeNull()
  })

  it('no hearts 1', () => {
    expect(h.cardIsPresent("♥1")).toBeFalsy();
  })
  
  it('has spades, clubs, hearts and diams 2', () => {
    expect(h.cardIsPresent("♣2")).toBeTruthy();
    expect(h.cardIsPresent("♠2")).toBeTruthy();
    expect(h.cardIsPresent("♥2")).toBeTruthy();
    expect(h.cardIsPresent("♦2")).toBeTruthy();
  })

  it('has a card', () => {
    expect(container.querySelector('article')).not.toBeNull();
  })

  test('snapshot of main', () => {
    expect(container.querySelector('main').innerHTML).toMatchSnapshot();
  })

});
