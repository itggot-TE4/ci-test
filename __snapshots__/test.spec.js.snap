// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.html snapshot of main 1`] = `"<article class=\\"red\\"><header><p>♥</p><p>2</p></header><h1>♥</h1><footer><p>♥</p><p>2</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>3</p></header><h1>♥</h1><footer><p>♥</p><p>3</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>4</p></header><h1>♥</h1><footer><p>♥</p><p>4</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>5</p></header><h1>♥</h1><footer><p>♥</p><p>5</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>6</p></header><h1>♥</h1><footer><p>♥</p><p>6</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>7</p></header><h1>♥</h1><footer><p>♥</p><p>7</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>8</p></header><h1>♥</h1><footer><p>♥</p><p>8</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>9</p></header><h1>♥</h1><footer><p>♥</p><p>9</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>10</p></header><h1>♥</h1><footer><p>♥</p><p>10</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>J</p></header><h1>♥</h1><footer><p>♥</p><p>J</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>D</p></header><h1>♥</h1><footer><p>♥</p><p>D</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>K</p></header><h1>♥</h1><footer><p>♥</p><p>K</p></footer></article><article class=\\"red\\"><header><p>♥</p><p>A</p></header><h1>♥</h1><footer><p>♥</p><p>A</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>2</p></header><h1>♣</h1><footer><p>♣</p><p>2</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>3</p></header><h1>♣</h1><footer><p>♣</p><p>3</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>4</p></header><h1>♣</h1><footer><p>♣</p><p>4</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>5</p></header><h1>♣</h1><footer><p>♣</p><p>5</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>6</p></header><h1>♣</h1><footer><p>♣</p><p>6</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>7</p></header><h1>♣</h1><footer><p>♣</p><p>7</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>8</p></header><h1>♣</h1><footer><p>♣</p><p>8</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>9</p></header><h1>♣</h1><footer><p>♣</p><p>9</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>10</p></header><h1>♣</h1><footer><p>♣</p><p>10</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>J</p></header><h1>♣</h1><footer><p>♣</p><p>J</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>D</p></header><h1>♣</h1><footer><p>♣</p><p>D</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>K</p></header><h1>♣</h1><footer><p>♣</p><p>K</p></footer></article><article class=\\"black\\"><header><p>♣</p><p>A</p></header><h1>♣</h1><footer><p>♣</p><p>A</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>2</p></header><h1>♦</h1><footer><p>♦</p><p>2</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>3</p></header><h1>♦</h1><footer><p>♦</p><p>3</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>4</p></header><h1>♦</h1><footer><p>♦</p><p>4</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>5</p></header><h1>♦</h1><footer><p>♦</p><p>5</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>6</p></header><h1>♦</h1><footer><p>♦</p><p>6</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>7</p></header><h1>♦</h1><footer><p>♦</p><p>7</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>8</p></header><h1>♦</h1><footer><p>♦</p><p>8</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>9</p></header><h1>♦</h1><footer><p>♦</p><p>9</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>10</p></header><h1>♦</h1><footer><p>♦</p><p>10</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>J</p></header><h1>♦</h1><footer><p>♦</p><p>J</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>D</p></header><h1>♦</h1><footer><p>♦</p><p>D</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>K</p></header><h1>♦</h1><footer><p>♦</p><p>K</p></footer></article><article class=\\"red\\"><header><p>♦</p><p>A</p></header><h1>♦</h1><footer><p>♦</p><p>A</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>2</p></header><h1>♠</h1><footer><p>♠</p><p>2</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>3</p></header><h1>♠</h1><footer><p>♠</p><p>3</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>4</p></header><h1>♠</h1><footer><p>♠</p><p>4</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>5</p></header><h1>♠</h1><footer><p>♠</p><p>5</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>6</p></header><h1>♠</h1><footer><p>♠</p><p>6</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>7</p></header><h1>♠</h1><footer><p>♠</p><p>7</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>8</p></header><h1>♠</h1><footer><p>♠</p><p>8</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>9</p></header><h1>♠</h1><footer><p>♠</p><p>9</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>10</p></header><h1>♠</h1><footer><p>♠</p><p>10</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>J</p></header><h1>♠</h1><footer><p>♠</p><p>J</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>D</p></header><h1>♠</h1><footer><p>♠</p><p>D</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>K</p></header><h1>♠</h1><footer><p>♠</p><p>K</p></footer></article><article class=\\"black\\"><header><p>♠</p><p>A</p></header><h1>♠</h1><footer><p>♠</p><p>A</p></footer></article>"`;
